Sources/AuthFoundation/User Management/Internal/CredentialCoordinatorImpl.swift

```diff
@@ -165,28 +187,31 @@ final class CredentialCoordinatorImpl: CredentialCoordinator {
         return nil
     }
 
+    @CredentialActor
     func resetToDefault() {
-        lock.withLock {
-            _tokenStorage = nil
-            _credentialDataSource = nil
-        }
+        _tokenStorage = nil
+        _credentialDataSource = nil
     }
 
-    func observe(oauth2 client: OAuth2Client) {
+    nonisolated func observe(oauth2 client: OAuth2Client) {
         client.add(delegate: self)
     }
 }
 
 extension CredentialCoordinatorImpl: OAuth2ClientDelegate {
-    func oauth(client: OAuth2Client, didRefresh token: Token, replacedWith newToken: Token?) {
+    nonisolated func oauth(client: OAuth2Client, didRefresh token: Token, replacedWith newToken: Token?) {
```

Comment:
Why nonisolated if `withAsyncGroup` guarantees isolation?

Reply:
It needs to be marked as `nonisolated` because the OAuth2ClientDelegate protocol is not marked as `Sendable`, so we need to explicitly assign `nonisolated` to the function declaration since CredentialCoordinatorImpl _is_ `Sendable` (due to its conformance to `CredentialCoordinator`).